### DISCUSSION TITLE
Complete Recovery docs out of date

### DISCUSSION THREAD

**Discussion by Big-Brother-Man** [ory/.github#54] [Q&A] [answered]
Tl;dr : The docs for complete recovery flow is out of sync.

Hi all, I hope you all are doing well.

I finished the recovery flow, I have noticed that the docs are out of sync for this one.

![image](https://user-images.githubusercontent.com/120871736/224572656-84cdbf11-6234-41fc-aa90-66c519f6325c.png)

If you follow the above and email you will run into infinite re-send 

This is because email is not actually part of the complete recovery. I was able to confirm this with the pre-build UI provided by ory cloud.

Here is a valid complete flow logic (without email)
```
  cookie:cookie,
      flow: formData.flowId,
      updateRecoveryFlowBody: { //This is needed ory needs to update document
        code: formData.code,
        method: "code",
        csrf_token: formData.csrf,
      },
    }
```

- [ ] Update the updateRecoveryFlowBody to not contain `email` property 
- [ ] Update the document to remove email
- [ ] Update @Ory/Client to propagate the changes.

**Comment by Benehiko** [comment, marked as answer]
Hi @Big-Brother-Man 

The recovery flow is submitted twice, once to get the email sent out to the email, and the second to submit the code sent out from the email.

The flow kind of looks a bit like this:

1. Create recovery flow
2. Submit email
3. Re-render form based on response (or get flow data through ID)
4. Submit code
5. Get redirected to settings flow on success OR
6. handle form error

**Reply by Benehiko** [reply]
So even though the documentation is a bit confusing here, it is difficult to show that the recovery actually has two states with the `code` strategy.

The `link` strategy for example only has 1 submit.

**Reply by Big-Brother-Man** [reply]
Thank you makes sense I was bit confused as to why email was not required since the body required it. Maybe this is a chance to clarify the docs on the two parts i.e. provide the email not code and provide code not email.

**Reply by Big-Brother-Man** [reply]
Either way kindly thank you for your answer!

**Reply by Benehiko** [reply]
I have created an issue since we definitely need to update this https://github.com/ory/docs/issues/1321
